go-controller/pkg/node/controllers/egressservice/egressservice.go

```diff
@@ -0,0 +1,927 @@
+package egressservice
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	egressserviceapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressservice/v1"
+	egressserviceinformer "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressservice/v1/apis/informers/externalversions/egressservice/v1"
+	egressservicelisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressservice/v1/apis/listers/egressservice/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	nodeipt "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/iptables"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	discoverylisters "k8s.io/client-go/listers/discovery/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+const (
+	Chain          = "OVN-KUBE-EGRESS-SVC" // called from nat-POSTROUTING
+	IPRulePriority = 5000                  // the priority of the ip rules created by the controller
+)
+
+type Controller struct {
+	stopCh <-chan struct{}
+	sync.Mutex
+	returnMark string // packets coming with this mark should not be modified
+	thisNode   string // name of the node we're running on
+
+	egressServiceLister egressservicelisters.EgressServiceLister
+	egressServiceSynced cache.InformerSynced
+	egressServiceQueue  workqueue.RateLimitingInterface
+
+	serviceLister  corelisters.ServiceLister
+	servicesSynced cache.InformerSynced
+
+	endpointSliceLister  discoverylisters.EndpointSliceLister
+	endpointSlicesSynced cache.InformerSynced
+
+	services map[string]*svcState // svc key -> state
```

Comment:
tbh I feel that they are quite interchangable but I lean towards keeping it services because it maps to a svcState (with lb,eps.. which are service things). wdyt?

Reply:
I am ok with that